rust/numaflow-core/src/monovertex/metrics.rs

```diff
@@ -75,55 +81,68 @@ pub(crate) struct UserDefinedContainerState {
 
 /// The global register of all metrics.
 #[derive(Default)]
-pub struct GlobalRegistry {
+struct GlobalRegistry {
     // It is okay to use std mutex because we register each metric only one time.
-    pub registry: parking_lot::Mutex<Registry>,
+    registry: parking_lot::Mutex<Registry>,
 }
 
 impl GlobalRegistry {
     fn new() -> Self {
         GlobalRegistry {
             // Create a new registry for the metrics
-            registry: parking_lot::Mutex::new(Registry::default()),
+            registry: parking_lot::Mutex::new(Registry::with_prefix(GLOBAL_PREFIX)),
         }
     }
 }
 
-/// GLOBAL_REGISTER is the static global registry which is initialized
-// only once.
-static GLOBAL_REGISTER: OnceLock<GlobalRegistry> = OnceLock::new();
+/// GLOBAL_REGISTRY is the static global registry which is initialized only once.
+static GLOBAL_REGISTRY: OnceLock<GlobalRegistry> = OnceLock::new();
 
-/// global_registry is a helper function to get the GLOBAL_REGISTER
+/// global_registry is a helper function to get the GLOBAL_REGISTRY
 fn global_registry() -> &'static GlobalRegistry {
-    GLOBAL_REGISTER.get_or_init(GlobalRegistry::new)
+    GLOBAL_REGISTRY.get_or_init(GlobalRegistry::new)
 }
 
-// TODO: let's do sub-registry for forwarder so tomorrow we can add sink and source metrics.
 /// MonoVtxMetrics is a struct which is used for storing the metrics related to MonoVertex
 // These fields are exposed as pub to be used by other modules for
 // changing the value of the metrics
 // Each metric is defined as family of metrics, which means that they can be
 // differentiated by their label values assigned.
-// The labels are provided in the form of Vec<(String, String)
+// The labels are provided in the form of Vec<(String, String)>
 // The second argument is the metric kind.
 pub struct MonoVtxMetrics {
     // counters
     pub read_total: Family<Vec<(String, String)>, Counter>,
     pub read_bytes_total: Family<Vec<(String, String)>, Counter>,
     pub ack_total: Family<Vec<(String, String)>, Counter>,
```

Comment:
I didn't move these to a sub-registry since they don't have a common prefix like `source_`